tensorflow/lite/experimental/micro/examples/micro_recognition/disco/image_util.cc

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/experimental/micro/examples/micro_recognition/disco/image_util.h"
+
+void ResizeConvertImage(tflite::ErrorReporter* error_reporter,
+                        int in_frame_width, int in_frame_height,
+                        int num_in_channels, int out_frame_width,
+                        int out_frame_height, int channels, uint8_t* in_image,
+                        uint8_t* out_image) {
+  // offset so that only the center part of rectangular image is selected for
+  // resizing
+  int width_offset = ((in_frame_width - in_frame_height) / 2) * num_in_channels;
+
+  int yresize_ratio = (in_frame_height / out_frame_height) * num_in_channels;
+  int xresize_ratio = (in_frame_width / out_frame_width) * num_in_channels;
```

Comment:
you get `int` if you use `int / int * int` and there exist some numerical problems.
is this what you expect?